src/solver/conv_asm_implicit_gemm_gtc_fwd_nhwc.cpp

```diff
@@ -875,6 +897,19 @@ bool ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC::IsApplicable(
                                    miopen::GetTypeSize(problem.GetInDataType())))
         return false;
 
+    {
+        auto largest_config = problem.IsFp32()
+                                  ? GetFwdXdlopsNHWCConfigLargestTileFp32()
+                                  : (problem.IsFp16() ? GetFwdXdlopsNHWCConfigLargestTileFp16()
+                                                      : GetFwdXdlopsNHWCConfigLargestTileBf16());
+        size_t current_block_size, current_grid_size, current_splits_4G;
+        std::tie(current_block_size, current_grid_size, current_splits_4G) =
```

Comment:
@CAHEK7 @carlushuang I am still wondering why we use tuples as return types, which is error prone (sensitive to the order) and to enforces us to use comments to keep the code understandable (like this):
```C++
static std::tuple<size_t, // block_size
                  size_t, // grid_size
                  size_t> // splits_4G
GetImplicitGemmGtcDynamicFwdXdlopsNHWCKernel(
    const ProblemDescription& problem,
    const PerformanceConfigAsmImplicitGemmGTCFwdXdlopsNHWC& config);
```
instead of structures that contain members with "normal" names.

Wrt binding -- it is better sometimes, but looks as another crutch to me.